### PR TITLE
Feature/mafia vote

### PR DESCRIPTION
**Pull request by davidlee04**

### CODE REVIEW

File: module/mafiaVote.js [hunk 1]
```diff
@@ -25,69 +24,99 @@ module.exports = {
...
-                    if(hasUser) {
+                if(targetExists) {
+                    if(splitMsg.length>=2) {
+                        //check to see if target==mafia
                         for(let i = 0; i < mafiaGames["mafia"].length; i++) {
-                            if(mafiaGames.players[user] == mafiaGames.mafia[i]) {
-                                msg.channel.send('You can\'t vote to kill other mafia.');
+                            if(targetID == mafiaGames.mafia[i]) {
+                                msg.author.dmChannel.send('You can\'t vote to kill other mafia.');
+                                //msg.reply('You can\'t vote to kill other mafia.');
```
Comment (LePhil) [inline module/mafiaVote.js]:
```suggestion
```

File: module/mafiaVote.js [hunk 2]
```diff
@@ -129,63 +164,62 @@ module.exports = {
...
+                    delete mafiaGames.userIdMap[key];
                 }
             }
-            for(let i = 0; i<mafiaGames["medics"].length; i++) {
-                if(mafiaGames["medics"][i] == voteCandidates[0]) {
-                    mafiaGames["medics"].splice(i, 1);
-                }
-            }*/
+            Object.values(mafiaGames["players"]).forEach(element => {
+                msg.client.fetchUser(element).then(u=> {
```
Comment (LePhil) [inline module/mafiaVote.js]: y u no use your fancy `sendDM` method?

File: module/mafiaVote.js [hunk 3]
```diff
@@ -103,21 +132,27 @@ module.exports = {
...
+        
+        //checking which candidates have highest votes
         for(let i = 0; i < voteValues.length; i++) {
             if(voteValues[i] > highestVote) {
                 highestVote = voteValues[i];
             }
         }
         for(let i = 0; i < voteValues.length; i++) {
-            if(voteValues[i] = highestVote) {
+            if(voteValues[i] == highestVote) {
```
Comment (LePhil) [inline module/mafiaVote.js]: nice

File: module/mafiaVote.js [hunk 4]
```diff
@@ -25,69 +24,99 @@ module.exports = {
...
             } else {
-                msg.channel.send('You can\'t run this command during the day.');
+                msg.author.dmChannel.send('You can\'t run this command during the day.');
+                //sendDM(msg, msg.author.id, 'You can\'t run this command during the day.');
+                //msg.reply('You can\'t run this command during the day.');
+                return;
             }
         } else {
-            msg.channel.send('You must be a mafia to run this command!');
+            msg.author.dmChannel.send('You must be a mafia to run this command!');
```
Comment (LePhil) [inline module/mafiaVote.js]: y u no use your fancy `sendDM` method?

File: module/mafiaVote.js [hunk 5]
```diff
@@ -25,69 +24,99 @@ module.exports = {
...
                 } else {
-                    msg.channel.send('Syntax: +mvote username');
+                    msg.author.dmChannel.send('Couldn\'t find user.');
+                    //sendDM(msg, msg.author.id, 'Couldn\'t find user.');
                     return;
                 }
             } else {
-                msg.channel.send('You can\'t run this command during the day.');
+                msg.author.dmChannel.send('You can\'t run this command during the day.');
+                //sendDM(msg, msg.author.id, 'You can\'t run this command during the day.');
```
Comment (LePhil) [inline module/mafiaVote.js]: y u no use your fancy `sendDM` method?

File: module/mafiaVote.js [hunk 6]
```diff
@@ -25,69 +24,99 @@ module.exports = {
...
+
+                        //handle votecounts
                         if(mafiaGamesVote["numVotes"]==mafiaGames["mafia"].length) {
                             this.handleVoteCount(msg);
                             return;
                         }
                     } else {
-                        msg.channel.send('User not found.');
+                        msg.author.dmChannel.send('Syntax: +mvote username');
+                        //sendDM(msg, msg.author.id, 'Syntax: +mvote username');
```
Comment (LePhil) [inline module/mafiaVote.js]: y u no use your fancy `sendDM` method?

File: module/mafiaVote.js [hunk 7]
```diff
@@ -25,69 +24,99 @@ module.exports = {
...
+
+                        //feedback to tell who the mafia voted for
+                        let authorName = "";
+                        for(var key in mafiaGames["userIdMap"]) {
+                            if(key == msg.author.id) {
+                                authorName = mafiaGames.userIdMap[key];
+                            }
+                        }
+                        //sendDM(msg, msg.author.id, mafiaGames.userIdMap[msg.author.id]+', you just voted for '+target);
+                        msg.author.dmChannel.send(mafiaGames.userIdMap[msg.author.id]+', you just voted for '+target);
```
Comment (LePhil) [inline module/mafiaVote.js]: y u no use your fancy `sendDM` method?

File: module/mafiaVote.js [hunk 8]
```diff
@@ -25,69 +24,99 @@ module.exports = {
...
-                    }
-                    if(hasUser) {
+                if(targetExists) {
+                    if(splitMsg.length>=2) {
+                        //check to see if target==mafia
                         for(let i = 0; i < mafiaGames["mafia"].length; i++) {
-                            if(mafiaGames.players[user] == mafiaGames.mafia[i]) {
-                                msg.channel.send('You can\'t vote to kill other mafia.');
+                            if(targetID == mafiaGames.mafia[i]) {
+                                msg.author.dmChannel.send('You can\'t vote to kill other mafia.');
```
Comment (LePhil) [inline module/mafiaVote.js]: y u no use your fancy `sendDM` method?

**Comment by EpicProgrammer207** [comment]
im gonna say the n word

**Pull request merged**